packages/nouns-contracts/test/foundry/governance/NounsDAOExecutorV3.t.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.19;
+
+import 'forge-std/Test.sol';
+import { DeployUtilsExcessETH } from '../helpers/DeployUtilsExcessETH.sol';
+import { NounsDAOExecutorV3 } from '../../../contracts/governance/NounsDAOExecutorV3.sol';
+import { IExcessETH } from '../../../contracts/interfaces/IExcessETH.sol';
+import { NounsDAOLogicV3 } from '../../../contracts/governance/NounsDAOLogicV3.sol';
+import { NounsDAOExecutorV2 } from '../../../contracts/governance/NounsDAOExecutorV2.sol';
+import { NounsTokenLike } from '../../../contracts/governance/NounsDAOInterfaces.sol';
+import { ExcessETH, INounsAuctionHouseV2 } from '../../../contracts/governance/ExcessETH.sol';
+import { NounsAuctionHouseProxyAdmin } from '../../../contracts/proxies/NounsAuctionHouseProxyAdmin.sol';
+import { NounsAuctionHouseProxy } from '../../../contracts/proxies/NounsAuctionHouseProxy.sol';
+import { NounsAuctionHouseV2 } from '../../../contracts/NounsAuctionHouseV2.sol';
+
+contract ExcessETHMock is IExcessETH {
+    uint256 excess;
+
+    function setExcess(uint256 excess_) public {
+        excess = excess_;
+    }
+
+    function excessETH() public view returns (uint256) {
+        return excess;
+    }
+}
+
+contract NounsDAOExecutorV3Test is DeployUtilsExcessETH {
+    event ETHBurned(uint256 amount);
+
+    NounsDAOExecutorV3 treasury;
+    ExcessETHMock excessETH;
+
+    address dao = makeAddr('dao');
+
+    function setUp() public {
+        treasury = _deployExecutorV3(dao);
+        excessETH = new ExcessETHMock();
+
+        vm.prank(address(treasury));
+        treasury.setExcessETHHelper(excessETH);
+    }
+
+    function test_setExcessETHHelper_revertsForNonTreasury() public {
+        vm.expectRevert('NounsDAOExecutor::setExcessETHHelper: Call must come from NounsDAOExecutor.');
+        treasury.setExcessETHHelper(excessETH);
+    }
+
+    function test_setExcessETHHelper_worksForTreasury() public {
+        ExcessETHMock newExcessETH = new ExcessETHMock();
+
+        assertTrue(address(treasury.excessETHHelper()) != address(newExcessETH));
+
+        vm.prank(address(treasury));
+        treasury.setExcessETHHelper(newExcessETH);
+
+        assertEq(address(treasury.excessETHHelper()), address(newExcessETH));
+    }
+
+    function test_burnExcessETH_givenHelperNotSet_reverts() public {
+        vm.prank(address(treasury));
+        treasury.setExcessETHHelper(IExcessETH(address(0)));
+
+        vm.expectRevert(abi.encodeWithSelector(NounsDAOExecutorV3.ExcessETHHelperNotSet.selector));
```

Comment:
@eladmallel nit: you don't need to wrap with `abi.encodeWithSelector`.
this:
`vm.expectRevert(NounsDAOExecutorV3.ExcessETHHelperNotSet.selector);`
should work

Reply:
thanks, fixed!